tests/test_basis.py

```diff
@@ -4523,6 +4342,306 @@ def test_power_of_basis(exponent, basis_class):
         )
 
 
+@pytest.mark.parametrize(
+    "basis_cls",
+    [
+        basis.MSplineBasis,
+        basis.BSplineBasis,
+        basis.CyclicBSplineBasis,
+        basis.RaisedCosineBasisLinear,
+        basis.RaisedCosineBasisLog,
+    ],
+)
+def test_basis_to_transformer(basis_cls):
+    n_basis_funcs = 5
+    bas = basis_cls(n_basis_funcs)
+    assert isinstance(bas.to_transformer(), basis.TransformerBasis)
+    assert bas.to_transformer().n_basis_funcs == bas.n_basis_funcs
```

Comment:
Do you mean something like this?

```python
public_attrs_basis = {attr for attr in dir(bas) if not attr.startswith("_")}
public_attrs_transformerbasis = {
    attr for attr in dir(bas.to_transformer()) if not attr.startswith("_")
}

assert public_attrs_transformerbasis - public_attrs_basis == {
    "fit",
    "fit_transform",
    "transform",
}

assert public_attrs_basis - public_attrs_transformerbasis == set()
```

Reply:
I used `__dict__` to test that the values of attributes are the same instead of just `n_basis_funcs`, and added another test to test the public names using the snippet above.